tests/core/string/test_string.h

```diff
@@ -350,18 +350,18 @@ TEST_CASE("[String] Substr") {
 
 TEST_CASE("[String] Find") {
 	String s = "Pretty Woman Woman";
-	CHECK(s.find("tty") == 3);
-	CHECK(s.find("Wo", 9) == 13);
-	CHECK(s.find("Revenge of the Monster Truck") == -1);
-	CHECK(s.rfind("man") == 15);
+	MULTICHECK_STRING_EQ(s, find, "tty", 3);
+	MULTICHECK_STRING_INT_EQ(s, find, "Wo", 9, 13);
+	MULTICHECK_STRING_EQ(s, find, "Revenge of the Monster Truck", -1);
+	MULTICHECK_STRING_EQ(s, rfind, "man", 15);
 }
 
 TEST_CASE("[String] Find no case") {
 	String s = "Pretty Whale Whale";
-	CHECK(s.findn("WHA") == 7);
-	CHECK(s.findn("WHA", 9) == 13);
-	CHECK(s.findn("Revenge of the Monster SawFish") == -1);
-	CHECK(s.rfindn("WHA") == 13);
+	MULTICHECK_STRING_EQ(s, findn, "WHA", 7);
+	MULTICHECK_STRING_INT_EQ(s, findn, "WHA", 9, 13);
+	MULTICHECK_STRING_EQ(s, findn, "Revenge of the Monster SawFish", -1);
+	MULTICHECK_STRING_EQ(s, rfindn, "WHA", 13);
```

Comment:
How about some more tests for finds. I would also suggest separating `find` and `rfind` test to separate test cases (also `findn` and `rfindn`).

These tests are still in the old format. 

```c++
	CHECK(s.find("") == 0);
	CHECK(s.find("", 9) == 9);
	CHECK(s.find("Pretty Woman Woman") == 0);
	CHECK(s.find("WOMAN") == -1);

	CHECK(s.rfind("") == 18); // fails (returns -1)
	CHECK(s.rfind("", 15) == 15); // fails (returns -1)
	CHECK(s.rfind("foo") == -1);
	CHECK(s.rfind("Pretty Woman Woman") == 0);
	CHECK(s.rfind("man", 13) == 9);
	CHECK(s.rfind("WOMAN") == -1);

	CHECK(s.findn("") == 0); // fails (returns -1)
	CHECK(s.findn("", 3) == 3); // fails (returns -1)
	CHECK(s.findn("wha") == 7);
	CHECK(s.findn("Wha") == 7);

	CHECK(s.rfindn("") == 15); // fails (returns -1)
	CHECK(s.rfindn("", 13) == 13); // fails (returns -1)
	CHECK(s.rfindn("wha") == 13);
	CHECK(s.rfindn("Wha") == 13);
```

There are some some inconsistencies how things work in master and thus some of these tests fail. Finding with "" seems to be a somewhat undefined case. The basic `find` returns `0` when `""` is searched, which to me seems to be the correct answer. There is a `""` at every index. But all other functions (`rfind`, `findn`, `rfindn`) return `-1` when `""` is searched. In my opinion that is wrong, when `""` is searched, I would expect the first index checked to be returned.

However, in GDScript both `find` and `rfind` return `-1` when "" is searched. Although I personally would prefer not returning `-1`, consistency is more important. If it is decided that `""` is never found (-1), current C++ `find` implementation should be changed.

Reply:
> The basic `find` returns `0` when `""` is searched, which to me seems to be the correct answer. There is a `""` at every index. But all other functions (`rfind`, `findn`, `rfindn`) return `-1` when `""` is searched.

That's a bug in the overload `String::find(const char *p_str, int p_from)`. It's documented it [should return `-1`](https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-find), and the same applies to `rfind`, `findn`, and `rfindn`. The way I see it, `""` is by definition an empty string and therefore searching for it implies nothing will be found and therefore the function should return `-1`.

I've patched this bug in this PR -- `find("")` now returns `-1` to match the behavior of the rest. Thanks for pointing that out.

> separate test cases

That's a style code change better suited for a different PR. I consider that change to be out of the scope of this PR.